In [2]:
import pandas as pd
import numpy as np

In [4]:
vine_df = pd.read_csv("vine_table.csv")
vine_df

review_id  star_rating  helpful_votes  total_votes vine  \
0         RTIS3L2M1F5SM            5              0            0    N   
1         R1ZV7R40OLHKD            5              0            0    N   
2        R3BH071QLH8QMC            1              0            1    N   
3        R127K9NTSXA2YH            3              0            0    N   
4        R32ZWUXDJPW27Q            4              0            0    N   
...                 ...          ...            ...          ...  ...   
1785992   RPC430LWZJ60T            5              1            1    N   
1785993  R347MZT5FH6HRJ            5              2            2    N   
1785994   RJ5BETZP0VIUS            1              0            2    N   
1785995   R85QTDO2KZMGO            1              1            3    N   
1785996  R2W36RZZZNTYWF            5              1            1    N   

        verified_purchase  
0                       Y  
1                       Y  
2                       Y  
3                       Y  
4                       Y  
...                   ...  
1785992                 N  
1785993                 N  
1785994                 N  
1785995                 N  
1785996                 N  

[1785997 rows x 6 columns]

In [5]:
high_votes_df = vine_df.loc[vine_df['total_votes'] > 20]
high_votes_df

review_id  star_rating  helpful_votes  total_votes vine  \
55        R4PKAZRQJJX14            1             21           34    N   
74       R2CI0Y288CC7E2            1             21           35    N   
209      R127WEQY2FM1T3            1            147          175    N   
289      R3EZ0EPYLDA34S            1             14           31    N   
483      R2FJ94555FZH32            2             55           60    N   
...                 ...          ...            ...          ...  ...   
1785641   RCMDCDJR16IKW            4             27           30    N   
1785688  R3GYUWLD9FWCPS            5             30           30    N   
1785714  R26KS4Q9G04FIV            2              8           21    N   
1785788   RU0J1ZMBCLD27            5             27           29    N   
1785935  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
55                      N  
74                      Y  
209                     Y  
289                     Y  
483                     N  
...                   ...  
1785641                 N  
1785688                 N  
1785714                 N  
1785788                 N  
1785935                 N  

[61153 rows x 6 columns]

In [6]:
helpful_votes_df = high_votes_df.loc[high_votes_df['helpful_votes']/high_votes_df['total_votes'] > 0.5]
helpful_votes_df

review_id  star_rating  helpful_votes  total_votes vine  \
55        R4PKAZRQJJX14            1             21           34    N   
74       R2CI0Y288CC7E2            1             21           35    N   
209      R127WEQY2FM1T3            1            147          175    N   
483      R2FJ94555FZH32            2             55           60    N   
537      R1U3AR67RE273L            1             51           65    N   
...                 ...          ...            ...          ...  ...   
1785614  R31UKJUAJNX7XX            5             35           50    N   
1785641   RCMDCDJR16IKW            4             27           30    N   
1785688  R3GYUWLD9FWCPS            5             30           30    N   
1785788   RU0J1ZMBCLD27            5             27           29    N   
1785935  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
55                      N  
74                      Y  
209                     Y  
483                     N  
537                     Y  
...                   ...  
1785614                 N  
1785641                 N  
1785688                 N  
1785788                 N  
1785935                 N  

[37475 rows x 6 columns]

In [13]:
paid_review_df = helpful_votes_df.loc[helpful_votes_df['vine'] == 'Y']
unpaid_review_df = helpful_votes_df.loc[helpful_votes_df['vine'] == 'N']

In [11]:
paid_review_df.count()      

review_id            90
star_rating          90
helpful_votes        90
total_votes          90
vine                 90
verified_purchase    90
dtype: int64

In [12]:
paid_review_df.loc[paid_review_df['star_rating'] == 5].count()

review_id            44
star_rating          44
helpful_votes        44
total_votes          44
vine                 44
verified_purchase    44
dtype: int64

In [9]:
unpaid_review_df.count()

review_id            37385
star_rating          37385
helpful_votes        37385
total_votes          37385
vine                 37385
verified_purchase    37385
dtype: int64

In [14]:
unpaid_review_df.loc[unpaid_review_df['star_rating'] == 5].count()

review_id            14626
star_rating          14626
helpful_votes        14626
total_votes          14626
vine                 14626
verified_purchase    14626
dtype: int64

90 helpful paid reviews

44 5* paid reviews

48.88% of paid reviews are 5*

37385 helpful unpaid reviews

14626 5* unpaid reviews

39.12% of unpaid reviews are 5*